src/libs/elektra/keyname.c

```diff
@@ -568,6 +568,142 @@ ssize_t keyAddName (Key * key, const char * newName)
 	return key->keySize;
 }
 
+static size_t replacePrefix (char ** buffer, size_t size, size_t oldPrefixSize, const char * newPrefix, size_t newPrefixSize)
+{
+	size_t newSize;
+	if (size == oldPrefixSize)
+	{
+		// overwrite everything
+		newSize = newPrefixSize;
+		elektraRealloc ((void **) buffer, newSize);
+		memcpy (*buffer, newPrefix, newPrefixSize);
+	}
+	else if (oldPrefixSize < newPrefixSize)
+	{
+		// grow, move, overwrite
+		newSize = size + (newPrefixSize - oldPrefixSize);
+		elektraRealloc ((void **) buffer, newSize);
+		memmove (*buffer + newPrefixSize, *buffer + oldPrefixSize, size - oldPrefixSize);
+		memcpy (*buffer, newPrefix, newPrefixSize);
+	}
+	else
+	{
+		// move, overwrite, shrink
+		newSize = size - (oldPrefixSize - newPrefixSize);
+		memmove (*buffer + newPrefixSize, *buffer + oldPrefixSize, size - oldPrefixSize);
+		memcpy (*buffer, newPrefix, newPrefixSize);
+		elektraRealloc ((void **) buffer, newSize);
+	}
+	return newSize;
+}
+
+/**
+ * Replaces a prefix of the key name of @p key.
+ *
+ * The function only modifies @p key, if is is below (or same as)
+ * @p oldPrefix (see keyIsBelowOrSame()) and they both have the
+ * same namespace (this is not always the case with keyIsBelowOrSame()).
+ *
+ * In simple terms this function operates as follows:
+ * 1. If before calling this function @p key and @p oldPrefix had the
+ *    same name, then afterwards @p key will have the same name as
+ *    @p newPrefix.
+ * 2. If @p key was in the same namespace as and below @p oldPrefix, then
+ *    after calling this function @p key will be in the same namespace as
+ *    and below @p newPrefix.
+ * 3. Otherwise @p key will not be modified.
+ *
+ * Note: We use `const Key *` arguments for the prefixes instead of
+ * `const char *` to ensure only valid key names can be passed as arguments.
+ *
+ * @param key       The key that will be manipulated.
+ * @param oldPrefix The name of this key will be removed from the front
+ *                  of the name of @p key.
+ * @param newPrefix The name of this key will be added to the front of
+ *                  @p key, after the name of @p oldPrefix is removed.
+ *
+ * @retval -1 if @p key, @p oldPrefix or @p newPrefix are NULL
+ *            or the name of @p key is marked as read-only
+ * @retval  0 if @p key is not below (or same as) @p oldPrefix,
+ *            i.e. there is no prefix to replace
+ * @retval  1 if the prefix was sucessfully replaced
+ */
+int keyReplacePrefix (Key * key, const Key * oldPrefix, const Key * newPrefix)
```

Comment:
I added this function to `kdb.h` since it is
1. quite useful to some plugins
2. much easier to use than manually calling `keyName`, `keySetName`, `keyNameSize` etc.
3. heavily optimized compared to the alternative
4. uses many internals of `struct _Key` that really shouldn't be modified outside the core